dev/crypto.md

```diff
@@ -477,6 +476,26 @@ if:
   
   T is defined in the [technical report][compactcert], section IV.
 
+## Setting security strength
+
+- ${target_C}$: "classical" security strength. This is set to ${k+q}$ (${k+q}$ are defined in section IV.A of the [technical report][compactcert]). The goal is to have ${<= 1/2^{k}}$ probability of breaking the state proof by an attacker that makes up to ${2^{q}}$ hash evaluations/queries. We use ${target_C}$ = 192, which corresponds to, for example, ${k=128}$ and ${q=64}$, or ${k=96}$ and ${q=96}$.
+- ${target_{PQ}}$: "post-quantum" security strength. This is set to ${k+2q}$, because at a cost of about ${2^q}$, a quantum attacker can search among up to ${2^{2q}}$ hash evaluations (this is a highly attacker-favorable estimate). We use ${target_{PQ} = 256}$, which corresponds to, for example, ${k=128}$ and ${q=64}$, or ${k=96}$ and ${q=80}$.
+
+## Bounding The Number of Reveals
+
+In order for the SNARK prover for State Proofs to be efficient enough, we must impose an upper bound ${MaxReveals_{C}}$ on the number of "reveals" the SP can contain, while still reaching its target security strength ${target_C = 192}$. Concretely, for now we wish to set  ${MaxReveals_{C} = 480}$.
+
+Similarly, the quantum-secure verifier aims for a larger security strength of ${target_{PQ} = 256}$, and we can also impose an upper bound ${MaxReveals_{PQ}}$ on the number of reveals it can handle. (Recall that a smaller number of reveals means that signedWeight/provenWeight must be larger in order to reach a particular security strength, so we cannot set ${MaxReveals_{C}}$ or ${MaxReveals_{PQ}}$ too low.)
+
+To generate a SNARK proof, we need to be able to "downgrade" a valid SP with ${target_{PQ}}$ strength into one with merely ${target_{C}}$ strength, by truncating some of the reveals to stay within the bounds.
+
+So, for a given ${MaxReveals_{C}}$ and the desired security strengths, we need to calculate a suitable ${target_{PQ}}$ bound so that the following property holds:
+
+Any valid SP with strength >= ${target_{PQ}}$ and <= ${MaxReveals_{PQ}}$ reveals can be truncated to a valid SP with strength >= ${target_{C}}$ and <= ${MaxReveals_{C}}$ reveals.
+
+According to the SNARK-friendly weight-verification formula used by the SP verifiers, using any ${MaxReveals_{PQ}}$ <= floor(${MaxReveals_{C}} * {target_{PQ}} / {target_C}$) guarantees that the above property holds. Since the quantum-secure verifier is not bottlenecked by reveals, we can take this to be an equality, i.e., ${MaxReveals_{PQ}}$ = floor(...). Therefore we need to set ${MaxReveals_{PQ}}$ to 640. 
```

Comment:
```suggestion
Let us now prove that ${MaxReveals_{PQ}}$ <= floor(${MaxReveals_{C}} * {target_{PQ}} / {target_C}$) guarantees that the above property holds. Indeed, in Claim 1.1 of the [SNARK-Friendly Weight Threshold Verification][weight-threshold] (which defines the formula used by the SP verifiers), the values d, b, T, Y, and D only depend on signedWeight, but not the number of reveals nor the target. Hence, assuming Equation (5) holds for ${NumReveals_{PQ}}$ and ${TargetReveals_{PQ}}$. Multiplying Equation (5) on each side by ${target_{C}}$ / ${target_{PQ}}$, we get that ${NumReveals_{PQ}} * {target_{C}} / {target_PQ}$ <= ${target_C}$ * T * Y / D. Hence, if ${NumReveals_{PQ}}$ <= ${MaxReveals_{PQ}}$ <= floor(${MaxReveals_{C}} * {target_{PQ}} / {target_C}$), then Equation (5) holds for ${NumReveals_{C}}$ and ${TargetReveals_{C}}$, which was what we wanted to demonstrate (note that Equation (4) does not depend on NumReveals nor TargetReveals).

 Since the quantum-secure verifier is not bottlenecked by reveals, we can take ${MaxReveals_{PQ}}$ <= floor(${MaxReveals_{C}} * {target_{PQ}} / {target_C}$) to be an equality, i.e., ${MaxReveals_{PQ}}$ = floor(...). Therefore we need to set ${MaxReveals_{PQ}}$ to 640. 
```